.circleci/config.yml

```diff
@@ -11,15 +8,23 @@ executors:
      architecture: "amd64"
      platform: "linux/amd64"
 
-
 jobs:
   lint:
-    executor: python/default
+    docker:
+      - image: python:3.8.18-slim-bullseye
     steps:
       - checkout
       - run:
-          name: Install python dependencies
-          command: pip install -r dev-requirements.txt
+          name: Install deps
+          command: apt update && apt upgrade && apt install -y curl gcc libc-dev libffi-dev
+      - run:
+          name: Install poetry
+          command: "curl -sSL https://install.python-poetry.org | python3 - && \ 
+              echo 'export PATH=/root/.local/bin:${PATH}' >> \"$BASH_ENV\" && \
+              source \"$BASH_ENV\""
```

Comment:
As per https://python-poetry.org/docs/#ci-recommendations, I think we should:
- lock to a specific version of poetry
- install it elsewhere so that we don't have to update PATH

Same in the `docker-compose-test.yaml` file